### PR TITLE
Change the default oblique angle from 20deg to 14deg

### PR DESCRIPTION
**Pull request by Ahmad-S792**
#### 28de25b653f64ca3fc8e87fbf1d1a963329a5342
<pre>
Change the default oblique angle from 20deg to 14deg

Change the default oblique angle from 20deg to 14deg
<a href="https://bugs.webkit.org/show_bug.cgi?id=248258">https://bugs.webkit.org/show_bug.cgi?id=248258</a>

Reviewed by Myles C. Maxfield.

This patch is to align Webkit with Blink / Chromium, Gecko / Firefox and Web-Specification. This is also part of Interop 2023 proposal [1].

[1] <a href="https://github.com/web-platform-tests/interop/issues/64">https://github.com/web-platform-tests/interop/issues/64</a>

Merge - <a href="https://chromium.googlesource.com/chromium/src/+/8bfd7d5770ec70f47e84b5cf33961b437a87c5a7">https://chromium.googlesource.com/chromium/src/+/8bfd7d5770ec70f47e84b5cf33961b437a87c5a7</a>

Web-Specification - <a href="http://w3c.github.io/csswg-drafts/css-fonts-4/#valdef-font-style-oblique-angle-90deg-90deg">http://w3c.github.io/csswg-drafts/css-fonts-4/#valdef-font-style-oblique-angle-90deg-90deg</a>

The specs refer to have default angle as 14deg - &quot;The lack of an &lt;angle&gt; represents 14deg.&quot;.

The imported test cases are leveraging font from Stephen Nixon using the Open Font License [2].

[2] <a href="https://github.com/arrowtype/vf-slnt-test/blob/main/fonts/license-recursive-OFL.txt#L3">https://github.com/arrowtype/vf-slnt-test/blob/main/fonts/license-recursive-OFL.txt#L3</a>

* Source/WebCore/platform/graphics/FontSelectionAlgorithm.h: Update &quot;italicThreshold&quot; and &quot;normalItalicValue&quot; from 20deg to 14deg
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/variations/slnt-variable.html: Added Test Case
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/variations/slnt-variable-ref.html: Added Test Case Reference
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/variations/slnt-variable-expected.html: Added Test Case Expectations
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/variations/slnt-backslant-variable.html: Added Test Case
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/variations/slnt-backslant-variable-ref.html: Added Test Case Reference
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/variations/slnt-backslant-variable-expected.html: Added Test Case Expectations
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/variations/at-font-face-matching-expected.txt: Updated Test Expectations
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/variations/resources/FontStyleTest-slnt_backslant-VF.woff2 Added Test Font
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/variations/resources/FontStyleTest-slnt-VF.woff2: Added Test Font
* LayoutTests/fast/text/font-style-parse.html: Rebaselined
* LayoutTests/fast/text/font-style-parse-expected.txt: Rebaselined
* LayoutTests/platform/glib/TestExpectations: Add &quot;Failure&quot; Expectation with reference bug
* LayoutTests/platform/ios/TestExpectations: Add &quot;Failure&quot; Expectation with reference bug
* LayoutTests/platform/mac/TestExpectations: Add &quot;Failure&quot; Expectation with reference bug

Canonical link: <a href="https://commits.webkit.org/258722@main">https://commits.webkit.org/258722@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/91510fa5e7c1d719432925864cdd1a05a26ade98

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/102785 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/11905 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/35812 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/112043 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/172266 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/106751 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/12918 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/2814 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/95037 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/109720 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/108560 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/9918 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/93126 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/37563 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/91769 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/24646 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/79303 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/5360 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/26063 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/5509 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/2512 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/11528 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/45556 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5989 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/7252 "Built successfully") | | | 
| | | | |

### CODE REVIEW

File: LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/variations/font-style-parsing-expected.txt
Comment (nt1m) [inline LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/variations/font-style-parsing-expected.txt]: Do we ever want to return `oblique` without an angle given it means different things before/after this patch? For compat purposes, maybe: https://searchfox.org/wubkat/rev/fd9594cdf418745718908fd1869a116307a46e7a/Source/WebCore/css/FontSelectionValueInlines.h#123-124 needs to change to:

```
if (style.value() == italicValue() && axis == FontStyleAxis::ital)
    return CSSValueItalic;
```

Though I wonder what @litherum thinks?

**Comment by Ahmad-S792** [comment]
@nt1m - Any idea on why the tests ([slnt-variable.html](https://github.com/WebKit/WebKit/blob/main/LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/variations/slnt-variable.html) & [slnt-backslant-variable.html](https://github.com/WebKit/WebKit/blob/main/LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/variations/slnt-backslant-variable.html)) are failing while they are 1:1 from Blink patch and I did the related change? Do we need anything else? Thanks!

**Comment by nt1m** [comment]
> @nt1m - Any idea on why the tests ([slnt-variable.html](https://github.com/WebKit/WebKit/blob/main/LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/variations/slnt-variable.html) & [slnt-backslant-variable.html](https://github.com/WebKit/WebKit/blob/main/LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/variations/slnt-backslant-variable.html)) are failing while they are 1:1 from Blink patch and I did the related change? Do we need anything else? Thanks!

Sorry, I thought this would be a straightforward merge, but I think we may need more work on the font side. Myles can probably help when he gets back from vacation.

**Comment by Ahmad-S792** [comment]
> > @nt1m - Any idea on why the tests ([slnt-variable.html](https://github.com/WebKit/WebKit/blob/main/LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/variations/slnt-variable.html) & [slnt-backslant-variable.html](https://github.com/WebKit/WebKit/blob/main/LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/variations/slnt-backslant-variable.html)) are failing while they are 1:1 from Blink patch and I did the related change? Do we need anything else? Thanks!
> 
> Sorry, I thought this would be a straightforward merge, but I think we may need more work on the font side. Myles can probably help when he gets back from vacation.

No worries! I will tag @litherum and once he is back, he will be able to help. 👍🏼

**Comment by nt1m** [comment]
Myles, for more context, these tests cover https://arrowtype.github.io/vf-slnt-test/

**Comment by Ahmad-S792** [comment]
> Myles, for more context, these tests cover https://arrowtype.github.io/vf-slnt-test/

@nt1m - It seems that this [FIXME](https://github.com/WebKit/WebKit/blob/main/Source/WebCore/platform/graphics/FontSelectionAlgorithm.h#L385) might be related to this.

**Comment by litherum** [comment]
I wonder if https://bugs.webkit.org/show_bug.cgi?id=189483 is the cause of the failures 🤔

**Comment by litherum** [comment]
Right, that's an unrelated bug.

This patch is still a good one, even if it doesn't fix the unrelated bug too. And this patch is already tested. I think we should just mark the 2 tests as failing (with links to https://bugs.webkit.org/show_bug.cgi?id=189483) and land this.

File: LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/variations/font-style-parsing-expected.txt
Comment (Ahmad-S792) [reply LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/variations/font-style-parsing-expected.txt]: Happy to do change but will wait for Myles's input as well. :-)

Comment (litherum) [reply LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/variations/font-style-parsing-expected.txt]: The spec [says](http://w3c.github.io/csswg-drafts/css-fonts-4/#font-style-prop):

> Computed value: the keyword specified, plus angle in degrees if specified

I think that "if specified" answers the question. But I do agree that @nt1m has a good point for compatibility. I think we should match the spec as it stands, but also open a CSSWG issue to see if we can remove the "if specified" clause.

Comment (nt1m) [reply LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/variations/font-style-parsing-expected.txt]: The test in question is `oblique calc(10grad + 5deg)`, so it is specified. I guess this is more about the "serialize to shortest form" rule here. That becomes `oblique 14deg` and since `14deg` is default it is omitted (which I think shouldn't be the case, for compat reasons)

Comment (nt1m) [reply LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/variations/font-style-parsing-expected.txt]: Anyway, my suggested change should fix this.

**Review by litherum — APPROVED** [review]

File: Source/WebCore/css/FontSelectionValueInlines.h
Comment (nt1m) [inline Source/WebCore/css/FontSelectionValueInlines.h]: After the conversation in https://github.com/w3c/csswg-drafts/issues/8291, I think we may want to hold off this change for now. Sorry about that, the previous revision looked good.

Comment (Ahmad-S792) [reply Source/WebCore/css/FontSelectionValueInlines.h]: No worries, I will discard this change and push new commit.

Comment (Ahmad-S792) [reply Source/WebCore/css/FontSelectionValueInlines.h]: Pushed old build and it was already passing across all. I think we can merge it now. Let me know your thoughts before I add 'merge-queue'. Thanks!

**Pull request closed without merging**
